scrnaseq.Rmd

```diff
@@ -1761,8 +1768,56 @@ if (markers_found) {
 }
 ```
 
-### Heatmaps of differentially expressed genes {.tabset}
-#### All up- and down-regulated genes
+## Dotplots per cluster per sample 
```

Comment:
Conceptually, this part focuses on marker genes, i.e. the differences between the clusters. It may be a bit confusing if we now bring in samples.

Why not return to the old dot plots? This code can be used in the DEG section.

Reply:
This was a question that came up with a client, and I think it is very sensible. Surely, we talk about marker genes, but what cells actually express these marker genes? All cells in the cluster? Cells of sample 1, cells of sample 2? 

The old dotplot is still there, this part is in addition. Moreover, I changed the colour-scheme which was just confusing before.

Reply:
We can think of a more sensible header if you find this confusing.

Reply:
The marker genes are calculated for the entire dataset and identify the cell types.

I would bring the difference in the DEG section.

Reply:
I changed the colour scheme in this PR, so I guess you don't have the update yet. 

Can you please suggest a better place for this section? 

It is not about differences of samples. It is about justifying whether marker genes are actually expressed in all cells of a cluster, or just a part of the cells that actually belong to a particular sample.

Reply:
Yes how about we use this plot for the DEG analyses? Marker genes are used to identify a cluster based on their expression. Sample should not play a role here since cell type markers are universal.

Reply:
Also, the dot size in the plot should be relative to the total cell number of the sample in the cluster.

Reply:
Once we identified clusters and marker genes, we want to characterise the clusters as cell types. In this context, I think it is a perfectly sensible question to ask whether a marker gene is expressed in all cells of the cluster, or only in cells belonging to 1 or 2 specific samples. A marker gene might be expressed in only a part of a cluster and still come out as significant overall.

Reply:
If we don't find a solution to this, we can leave it open and discuss it in our bioinformatics meeting this week. The PR merge has to wait in that case.

Reply:
Hmm, okay, let us discuss this.

The PR do not have to wait. Just move the code in a separate branch.

Reply:
Nope, no additional messing around. My branch is called dotplot as I'd like to solve this. Everything is committed and pushed, so you could use it already and remove this part.

Reply:
Okay, then it will wait.

Reply:
How about we move "Dotplots per cluster per sample" towards the end of the Marker genes section, after functional enrichment analysis, and we call it "Expression per cluster and sample", and we only add it if we have at least 2 samples?